docs/api-reference/index.md

```diff
@@ -0,0 +1,21 @@
+# API Reference
+
+## Classes
+
+```{eval-rst}
+.. currentmodule:: {{projectname}}
+
+.. autosummary::
+   :toctree: ../generated/classes
+   :template: class-template.rst
+   :recursive:
+
+```
+
+## Top-level functions
+
+```{eval-rst}
+.. autosummary::
+   :toctree: ../generated/functions
+   :recursive:
+```
```

Comment:
What abut modules? We don't have them in all projects. But it is easier to remove something from the docs than adding it.